python/lsst/dax/apdb/cli/options.py

```diff
@@ -0,0 +1,114 @@
+# This file is part of dax_apdb
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+from __future__ import annotations
+
+__all__ = ["cassandra_config_options", "common_apdb_options", "sql_config_options"]
+
+import argparse
+from typing import TYPE_CHECKING, Any
+
+from ..apdb import ApdbConfig
+from ..apdbCassandra import ApdbCassandraConfig
+from ..apdbSql import ApdbSqlConfig
+
+if TYPE_CHECKING:
+    from lsst.pex.config import Field
+
+
+def _option_from_pex_field(
+    group: argparse._ArgumentGroup,
+    field: Field,
+    *,
+    name: str | None = None,
+    help: str | None = None,
+    **kwargs: Any,
+) -> None:
+    """Convert pex_config Field to argparse argument definition."""
+    if name is None:
+        name = "--" + field.name.replace("_", "-")
+    if help is None:
+        help = field.doc
+    group.add_argument(name, dest=field.name, help=help, **kwargs)
+
+
+# Options for fields in ApdbConfig.
+def common_apdb_options(parser: argparse.ArgumentParser) -> None:
+    """Define common configuration options."""
+    group = parser.add_argument_group("common APDB options")
+    _option_from_pex_field(group, ApdbConfig.schema_file, metavar="URL")
+    _option_from_pex_field(group, ApdbConfig.schema_name)
+    _option_from_pex_field(group, ApdbConfig.read_sources_months, type=int)
+    _option_from_pex_field(group, ApdbConfig.read_forced_sources_months, type=int)
+    _option_from_pex_field(group, ApdbConfig.use_insert_id, action="store_true", default=False)
```

Comment:
But we already have documentation (on pipelines.lsst.io) for the config classes, in a form that isn't limited by the constraints of a command line. By creating a new interface, aren't you basically forcing the user to wade through it by hand just in case it doesn't _quite_ match up with what's online?

Reply:
Situation for apdb-cli will be different than for usual task configs, there are at least two types of configs that are involved here - some parameters are needed only when you create database and some parameters are needed only when you make Apdb instances. There is no reason to keep a former set of options in the same config class (or maybe in any config class). I do want for the users to be able to say `apdb-cli ... --help` and see a correct syntax for every option, and not try to guess how exactly to pass a list of strings to some of the options.